src/lib/format/l20n/ast/parser.js

```diff
@@ -7,18 +7,20 @@ const MAX_PLACEABLES = 100;
 
 export default {
   parse: function(env, string, pos = false) {
-    this._source = string;
-    this._index = 0;
-    this._length = string.length;
-    this._curEntryStart = 0;
-    this._config = {
-      pos: pos
-    };
-
-    if (pos !== true) {
-      AST.Node.prototype.setPosition = function() {};
-    }
-    return this.getResource(pos);
+    AST.Node._config.pos = pos;
+
+    const parseContext = this.getNewParseContext(env, string, pos);
+
+    return parseContext.getResource();
+  },
+
+  getNewParseContext: function(env, string, pos = false) {
+    const parseContext = Object.create(this);
```

Comment:
Maybe all of the internal parsing functions should simply be methods on `ParseContext.prototype`?